src/app/space/create/pipelines/services/pipelines.service.spec.ts

```diff
@@ -309,4 +309,62 @@ describe('Pipelines Service', () => {
       );
   });
 
+  it('should emit response with repeat calls', (done: DoneFn) => {
+    const userServiceResponse = {
+      'data': {
+        'attributes': {
+          'created-at': '2017-05-11T16:44:56.376777Z',
+          'namespaces': [{
+            'cluster-app-domain': '8a09.starter-us-east-2.openshiftapps.com',
+            'cluster-console-url': 'https://console.starter-us-east-2.openshift.com/console/',
+            'cluster-logging-url': 'https://console.starter-us-east-2.openshift.com/console/',
+            'cluster-metrics-url': 'https://metrics.starter-us-east-2.openshift.com/',
+            'cluster-url': 'https://api.starter-us-east-2.openshift.com/',
+            'created-at': '2017-05-11T16:44:56.561538Z',
+            'name': 'blob',
+            'state': 'created',
+            'type': 'user',
+            'updated-at': '2017-05-11T16:44:56.561538Z',
+            'version': '1.0.91'
+          }]
+        },
+        'id': 'eae1de87-f58f-4e67-977a-95024dd7c6aa',
+        'type': 'userservices'
+      }
+    };
+
+    const subscription: Subscription = mockBackend.connections.subscribe((connection: MockConnection) => {
+      connection.mockRespond(new Response(
+        new ResponseOptions({
+          body: JSON.stringify(userServiceResponse),
+          status: 200
+        })
+      ));
+    });
+
+    svc.getOpenshiftConsoleUrl()
+      .subscribe(
+        (msg: string) => {
+          expect(msg).toEqual('https://console.starter-us-east-2.openshift.com/console/project/blob/browse/pipelines');
+          subscription.unsubscribe();
+          done();
```

Comment:
Is this test case working as intended? You're calling `done` and unsubscribing the mock connection handler in both subscription blocks, which seems wrong.

Reply:
Hmm good point. I'm not sure how I can get an unsubscribe in at the end of the test without encapsulating this in a describe with before/after Each/All. Any ideas?

Reply:
I've gone with that solution for now. If you have a better idea I'd be happy to incorporate it!